ipmininet/router/config/zebra.py

```diff
@@ -112,121 +122,156 @@ def __init__(self, name: Optional[str] = None, action=PERMIT,
+            0, ge, type_mask[self.family])
         if le is not None and ge is not None:
             assert le >= ge, "assertion le (%d) >= ge (%d) failed! le must be lower than ge" % (le, ge)
 
         self.le = le
         self.ge = ge
 
-        super().__init__(_prefix, action)
 
-
-class PrefixList:
-    # Number of PFXL
+class ZebraList(ABC):
     count = 0
 
-    def __init__(self, entries, family, name=None):
+    @property
+    @abstractmethod
+    def prefix_name(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def Entry(self):
+        raise NotImplementedError
+
+    def __init__(self, family, entries: Sequence[Union['ZebraList.Entry',
+                                                       str, IPv4Network,
+                                                       IPv6Network]] = (), name=None):
         assert family in {'ipv4', 'ipv6'}, "PrefixList unknown %s type. type must be either ipv4 or ipv6" % family
 
-        PrefixList.count += 1
+        ZebraList.count += 1
 
-        self.name = name if name else 'pfxl%d' % PrefixList.count
+        self.name = name if name else '%s%d' % (self.prefix_name, ZebraList.count)
         self.entries = []
         for e in entries:
-            if isinstance(e, PrefixListEntry):
-                assert e.family == family, "The prefix entry must be on the same type"
+            if isinstance(e, self.Entry):
+                assert e.family == family, "The prefix entry must be of the same type"
                 self.entries.append(e)
+            elif isinstance(e, str) and e == 'any':
+                self.entries.append(self.Entry(prefix=e, family=family))
+            elif isinstance(e, IPv4Network) or isinstance(e, IPv6Network) or isinstance(e, str):
+                self.entries.append(self.Entry(prefix=e))
             else:
-                ne = PrefixListEntry(prefix=e)
-                assert ne.family == family
-                self.entries.append(ne)
+                raise ValueError('"%s" is not a valid prefix entry for the %s family' % (e, family))
 
         self.family = family
 
     def __eq__(self, other):
         return self.name == other.name
 
 
+class PrefixList(ZebraList):
+    @property
+    def prefix_name(self):
+        return 'pfxl'
+
+    @property
+    def Entry(self):
+        return PrefixListEntry
+
+
+class AccessList(ZebraList):
+    """A zebra access-list class. It contains a set of AccessListEntry,
+    which describes all prefix belonging or not to this ACL"""
+
+    @property
+    def prefix_name(self):
+        return 'acl'
+
+    @property
+    def Entry(self):
+        return AccessListEntry
+
+
 class RouteMapMatchCond:
     """
     A class representing a RouteMap matching condition
     """
 
-    def __init__(self, cond_type: str, condition, family: Optional[str] = None):
+    def __init__(self, cond_type: str, condition, family=None):
```

Comment:
Why did you remove the type ?

Reply:
Oops, this this was due to an oversight in a local refactor. BTW changed in dca805acddbe19863aea03c4ed488785ca0b650e